### ISSUE TITLE
Example usage in help page

### ISSUE THREAD

**Issue by Lestropie** [MRtrix3/mrtrix3#1086]
Each command could have one or more examples of how the command could be used. This is particularly relevant for commands that don't have a fixed argument list; e.g. `dwi2fod msmt_csd`, `tckedit` (for concatenating track files), `mrmath` with wildcard input, some commands e.g. `mrcalc`, `transformcalc` basically already have this but in the `DESCRIPTION` field.

Up for debate as to whether this field should be compulsory.

**Comment by portokalh** [comment]
this would be very helpful IMHO. best!

**Comment by Lestropie** [comment]
Just received another +1 for this due to `meshfilter` being designed like `mrfilter`, but currently only having one filter option, hence the interface looking kinda weird.

`mrconvert` is another that would I think benefit from an explicit example usage section.

**Comment by jdtournier** [comment]
OK, that would require a bit of effort... Also, I'd suggest having a triplet, for operation, command, description, e.g:

---

## Examples

#### Extract every other volume from 4D dataset
```
$ mrconvert in.mif -coord 3 0:2:end out.mif
```
This makes use of number sequences and the 'end' keyword to select coordinates 0, 2, 4, ..., end along the volume axis (axis 3). 

#### Flip an image upside down
```
$ mrconvert in.mif -coord 2 end:0 out.mif
```
This makes use of number sequences and the 'end' keyword to select coordinates end, end-1, ... 1, 0 along the z-axis (axis 2), effectively flipping the image upside down. 

---

My only concern about this is that there are commands like `mrconvert` where the list of examples could grow _very_ large...

**Comment by Lestropie** [comment]
> OK, that would require a bit of effort

A bit, sure; but it's not all that different from other command interface functionalities that are already there.

> I'd suggest having a triplet

:+1:

> My only concern about this is that there are commands like mrconvert where the list of examples could grow very large

Well it's already at a decent length; it's just all shoe-horned into the description section. Which is better than previous, when each individual option had a massively-long description.

**Comment by jdtournier** [comment]
OK, let's do it. One more question though: where should the examples show up? Between description and options? Or after options? In most other software help pages, it tends to come after the options...

**Comment by thijsdhollander** [comment]
Naturally, I'd have thought of them appearing right after the description (i.e. before the options)... however, with this in mind:

> My only concern about this is that there are commands like mrconvert where the list of examples could grow very large...

...putting them right after the description may end up being quite frustrating to most non-beginner users, who wouldn't need the examples any more, but who do still frequently rely on the option list for reference (as I still often do myself, actually).  So well, taking that into account, I think it may be better to put them after the options.

**Comment by Lestropie** [comment]
Another :+1: coming for this from me personally. In the Python port of `foreach`, the example usage:
```
$ foreach folder/*.mif : mrinfo IN
```
results in a Sphinx warning because the wildcard character is interpreted in an RST file as an unterminated inline emphasis :sweat_smile:

**Comment by jdtournier** [comment]
closed with #1489
